build-coordinator/src/main/java/org/jboss/pnc/core/builder/BuildTask.java

```diff
@@ -99,7 +99,7 @@ public void setStatus(BuildStatus status) {
         log.debug("Updating build task {} status to {}", this.getId(), buildStatusChanged);
         buildSetTask.taskStatusUpdated(buildStatusChanged);
         buildStatusChangedEvent.fire(buildStatusChanged);
-        if (status.equals(BuildStatus.DONE)) {
```

Comment:
Putting it onther way around will save you from NPE `BuildStatus.DONE.equals(status)`